Tests/MongoSwiftTests/APMTests.swift

```diff
@@ -0,0 +1,240 @@
+            expect(eventTypes.count).to(beGreaterThan(0))
+        }
+        for i in 1...4 {
+            let eventTypesOld = await eventArray.eventTypes[i - 1]
+            let eventTypesCurr = await eventArray.eventTypes[i]
+            expect(eventTypesOld).to(equal(eventTypesCurr))
+        }
+    }
+
+    func testCommandStreamConcurrentStreams() async throws {
+        try await self.concurrencyHelper { client in
+            client.commandEventStream()
+        }
+    }
+
+    func testSDAMStreamConcurrentStreams() async throws {
+        try await self.concurrencyHelper { client in
+            client.sdamEventStream()
+        }
+    }
+
+    func testClientFinishesSDAMStreams() async throws {
+        let elg = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+        defer { elg.syncShutdownOrFail() }
+        let client = try MongoClient.makeTestClient(eventLoopGroup: elg)
+
+        // Task to create a commandEventStream and should return when the client is closed
+        let eventTask = Task { () -> Bool in
+            for try await _ in client.sdamEventStream() {}
+            return true
+        }
+        try await client.db("admin").runCommand(["ping": 1])
+        try await client.close()
+
+        // Ensure the task returned eventually
+        _ = try await eventTask.value
+    }
+
+    func testSDAMEventStreamHasCorrectEvents() async throws {
+        actor EventArray {
+            var streamEvents: [EventType] = []
+
+            func append(event: EventType) {
+                self.streamEvents.append(event)
+            }
+        }
+        // Need lock to prevent dataraces
+        let lock = Lock()
+        let taskStarted = NIOAtomic<Bool>.makeAtomic(value: false)
+        var handlerEvents: [EventType] = []
+        let eventArray = EventArray()
+        try await self.withTestClient { client in
+            client.addSDAMEventHandler { event in
+                lock.withLock {
+                    if !event.isHeartbeatEvent {
+                        handlerEvents.append(event.type)
+                    }
+                }
+            }
+            Task {
+                let sdamStream = client.sdamEventStream()
+                taskStarted.store(true)
+                for try await event in sdamStream {
+                    if !event.isHeartbeatEvent {
+                        await eventArray.append(event: event.type)
+                    }
+                }
+            }
+            // Wait until the event stream has been opened before we start inserting data.
+            try await assertIsEventuallyTrue(description: "event stream should be started") {
+                taskStarted.load()
+            }
+
+            try await client.db("admin").runCommand(["ping": 1])
+            try await client.db("trialDB").collection("trialColl").insertOne(["hello": "world"])
+        }
+        let streamEvents = await eventArray.streamEvents
+        expect(streamEvents).to(equal(handlerEvents))
```

Comment:
can we add an expectation that `streamEvents.count` is greater than 0? Just as a sanity check that the test works.

Reply:
Done. Good extra sanity test